apps/language_server/test/providers/completion_test.exs

```diff
@@ -77,4 +77,57 @@ defmodule ElixirLS.LanguageServer.Providers.CompletionTest do
              "def my_fun(example,arg)"
            ]
   end
+
+  test "returns module completions after pipe" do
+    text = """
+    defmodule MyModule do
+      NaiveDateTime.utc_now() |> Naiv
+    #                                ^
+    1..100
+    |> Enum.map(&Inte)
+    #                ^
+    def my(%Naiv)
+    #           ^
+    end
+    """
+
+    {line, char} = {1, 33}
+    TestUtils.assert_has_cursor_char(text, line, char)
+    {:ok, %{"items" => items}} = Completion.completion(text, line, char, true)
+
+    completions =
+      items
+      |> Enum.filter(&(&1["detail"] =~ "struct"))
+      |> Enum.map(& &1["label"])
+
+    assert completions == [
+             "NaiveDateTime"
```

Comment:
Relying on completion of modules in the stdlib is not ideal, but since the module names themselves are unlikely to change (unlike function names or docs) I think it's okay in this instance. Although it would probably be better to check that `NaiveDateTime` is one of the completion results instead of the only one in case a new module beginning with `Naiv` is added to the stdlib in the future.

Reply:
Good call, I changed it to check if the module in question is one of the results.